storage/ActiongraphImpl.cc

```diff
@@ -130,6 +130,23 @@ namespace storage
     }
 
 
+    void
+    Actiongraph::Impl::add_chain(const vector<vector<vertex_descriptor>>& vert_vectors)
+    {
+	vector<vector<Actiongraph::Impl::vertex_descriptor>> non_empty_vectors;
+	for (const vector<vertex_descriptor>& vector : vert_vectors)
+	    if (!vector.empty()) non_empty_vectors.push_back(vector);
+
+	if (non_empty_vectors.size() > 1)
+	    for (size_t i = 1; i < non_empty_vectors.size(); ++i)
```

Comment:
Hm... this has O(n^3) AFAICS. Probably n will always be pretty small, but can we be sure about that? What about S/390 that might have some hundred DASDs with up to 3 partitions on each? Will that hurt us?